pyluna-pathology/luna/pathology/cli/dsa/dsa_upload.py

```diff
@@ -8,94 +10,104 @@
+    required=False,
+    help="path to a metadata json/yaml file with method parameters to "
+    "reproduce results",
+)
+def cli(**cli_kwargs):
     """DSA Annotation Upload CLI
 
-    config - yaml file with DSA instance details:
-
-    - host: DSA host e.g. localhost
-
-    - port: DSA port e.g. 8080
-
-    - dsa_username: DSA username
-
-    - dsa_password: DSA password
-
-    data_config - yaml file with input, output, and method parameters:
-
-    - annotation_filepath: path to a DSA compatible annotation json file. often generated by dsa_viz CLI.
-
-    - image_filename: name of the image file in DSA e.g. 123.svs
-
-    - collection_name: name of the collection in DSA
+    Example:
+        export DSA_USERNAME=username
+        export DSA_PASSWORD=password
+        dsa_upload http://localhost:8080/dsa/api/v1
+            --collection_name tcga-data
+            --image_filename 123.svs
+            --annotation_name TumorVsOther
     """
-    with open(config, "r") as config_yaml:
-        dsa_config_dict = yaml.safe_load(config_yaml)
-
-    with open(data_config) as data_config_yaml:
-        data_config_dict = yaml.safe_load(data_config_yaml)
-
-    uri = dsa_config_dict["host"] + ":" + str(dsa_config_dict["port"])
-
-    dsa_username = dsa_config_dict["dsa_username"]
-    dsa_password = dsa_config_dict["dsa_password"]
-
-    gc = girder_client.GirderClient(apiUrl=f"http://{uri}/api/v1")
-    gc.authenticate(dsa_username, dsa_password)
-
-    # check DSA connection
-    system_check(gc)
-
-    annotation_filepath = data_config_dict["annotation_filepath"]
-    collection_name = data_config_dict["collection_name"]
-    image_filename = data_config_dict["image_filename"]
-
-    upload_annotation_to_dsa(
-        annotation_filepath, image_filename, collection_name, uri, gc
-    )
+    params = [
+        ("annotation_filepath", str),
+        ("collection_name", str),
+        ("image_filename", str),
+        ("dsa_endpoint", str),
+    ]
+    cli_runner(cli_kwargs, params, upload_annotation_to_dsa)
 
 
 def upload_annotation_to_dsa(
-    annotation_filepath, image_filename, collection_name, uri, gc
+    annotation_filepath, image_filename, collection_name, dsa_endpoint
 ):
     """Upload annotation to DSA
 
     Upload json annotation file as a new annotation to the image in the DSA collection.
 
     Args:
-        annotation_filepath (string): path to a DSA compatible annotation json file
+        annotation_filepath (string): path to a DSA annotation json file
         image_filename (string): name of the image file in DSA e.g. 123.svs
         collection_name (string): name of the collection in DSA
-        uri (string): DSA host:port e.g. localhost:8080
-        gc: girder client
+        uri (string): DSA API endpoint e.g. http://localhost:8080/api/v1
 
     Returns:
-        None
+        dict: item_uuid. None if item doesn't exist
     """
+    if os.environ["DSA_USERNAME"] is None or os.environ["DSA_PASSWORD"] is None:
```

Comment:
If you change the cli call to `cli(auto_envvar_prefix='DSA')`,  the username and password can be used as click arguments, see, https://github.com/msk-mind/luna/blob/c638f55a7d544dc29ca80797e2c0801ddf0079d7/pyluna-pathology/luna/pathology/cli/dsa_annotation_etl.py

Reply:
Thanks for the suggestion. I used the `auto_envvar_prefix` and added some filtering in the cli_runner, so we are not persisting possibly sensitive info  like username/pw. List of sensitive params are captured in commion/utils.py `MASK_KEYS`